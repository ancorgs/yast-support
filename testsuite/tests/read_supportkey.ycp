// testedfiles: supportio.ycp

{

    include "testsuite.ycp";

    include "support/supportio.ycp";

    map READ_ok_key =
    $[ "target":
        $[ "stat":
           $[ "size": 227]
        ],
       "support":
		$[ "key":
            $[ "KEY_NAME1": "Washington",
               "KEY_NAME2": "Irwing",
               "KEY_SALUTATION": "male",
               "KEY_COMPANY": "",
               "KEY_STREET": "",
               "KEY_ZIP": "",
               "KEY_COUNTRY": "",
               "KEY_STATE": "",
               "KEY_CITY": "",
               "KEY_EMAIL": "wi@iw.jh",
               "KEY_CODE": "WI-CATCH22-Yo-Yo"
            ]
        ]
	];

    map READ_ok_nokey =
    $[ "target":
        $[ "stat": $[]
        ],
       "support":
		$[ "key":
            $[ "KEY_NAME1": "",
               "KEY_NAME2": "",
               "KEY_SALUTATION": "",
               "KEY_COMPANY": "",
               "KEY_STREET": "",
               "KEY_ZIP": "",
               "KEY_COUNTRY": "",
               "KEY_STATE": "",
               "KEY_CITY": "",
               "KEY_EMAIL": "",
               "KEY_CODE": ""
            ]
        ]
	];

    map EX_ok =
    $[ "target":
        $[ "mkdir": true,
           "bash": 0
        ]
    ];

    map READ_nil_nokey =
    $[ "target":
        $[ "stat": $[]
        ],
       "support":
		$[ "key": nil ]
	];

    map EX_notok =
    $[ "target":
        $[ "mkdir": true,
           "bash": 1
        ]
    ];

    // keydata file present at usual place
	TEST (``(read_supportkey()), [ READ_ok_key, $[], $[] ], nil);
    DUMP ("----------------------");

    // keydata file not present, but successfully copied from template
	TEST (``(read_supportkey()), [ READ_ok_nokey, $[], EX_ok ], nil);
    DUMP ("----------------------");

    // keydata file cannot be copied and nil data will be read
	TEST (``(read_supportkey()), [ READ_nil_nokey, $[], EX_notok ], nil);

}
