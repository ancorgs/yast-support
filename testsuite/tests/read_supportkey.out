Read	.target.stat "/var/lib/YaST2/support/key" $["size":227]
Read	.support.key.KEY_NAME1 "Washington"
Read	.support.key.KEY_NAME2 "Irwing"
Read	.support.key.KEY_SALUTATION "male"
Read	.support.key.KEY_COMPANY ""
Read	.support.key.KEY_STREET ""
Read	.support.key.KEY_ZIP ""
Read	.support.key.KEY_COUNTRY ""
Read	.support.key.KEY_STATE ""
Read	.support.key.KEY_CITY ""
Read	.support.key.KEY_EMAIL "wi@iw.jh"
Read	.support.key.KEY_CODE "WI-CATCH22-Yo-Yo"
Return	$["city":"", "company":"", "country":"", "email":"wi@iw.jh", "key":"WI-CATCH22-Yo-Yo", "name1":"Washington", "name2":"Irwing", "salutation":"male", "state":"", "street":"", "zip":""]
Dump	----------------------
Read	.target.stat "/var/lib/YaST2/support/key" $[]
Execute	.target.mkdir "/var/lib/YaST2/support" true
Execute	.target.bash "/bin/cp /usr/share/YaST2/data/support/key /var/lib/YaST2/support/key" 0
Read	.support.key.KEY_NAME1 ""
Read	.support.key.KEY_NAME2 ""
Read	.support.key.KEY_SALUTATION ""
Read	.support.key.KEY_COMPANY ""
Read	.support.key.KEY_STREET ""
Read	.support.key.KEY_ZIP ""
Read	.support.key.KEY_COUNTRY ""
Read	.support.key.KEY_STATE ""
Read	.support.key.KEY_CITY ""
Read	.support.key.KEY_EMAIL ""
Read	.support.key.KEY_CODE ""
Return	$["city":"", "company":"", "country":"", "email":"", "key":"", "name1":"", "name2":"", "salutation":"", "state":"", "street":"", "zip":""]
Dump	----------------------
Read	.target.stat "/var/lib/YaST2/support/key" $[]
Execute	.target.mkdir "/var/lib/YaST2/support" true
Execute	.target.bash "/bin/cp /usr/share/YaST2/data/support/key /var/lib/YaST2/support/key" 1
Read	.support.key.KEY_NAME1 nil
Read	.support.key.KEY_NAME2 nil
Read	.support.key.KEY_SALUTATION nil
Read	.support.key.KEY_COMPANY nil
Read	.support.key.KEY_STREET nil
Read	.support.key.KEY_ZIP nil
Read	.support.key.KEY_COUNTRY nil
Read	.support.key.KEY_STATE nil
Read	.support.key.KEY_CITY nil
Read	.support.key.KEY_EMAIL nil
Read	.support.key.KEY_CODE nil
Return	$["city":nil, "company":nil, "country":nil, "email":nil, "key":nil, "name1":nil, "name2":nil, "salutation":nil, "state":nil, "street":nil, "zip":nil]
